pkg/controllers/deprovisioning/controller.go

```diff
@@ -298,33 +303,65 @@ func (c *Controller) waitForDeletion(ctx context.Context, nodeClaim *v1beta1.Nod
 	}
 }
 
-func (c *Controller) setNodesUnschedulable(ctx context.Context, isUnschedulable bool, candidates ...*Candidate) error {
+func (c *Controller) taintCandidates(ctx context.Context, addTaint bool, nodes ...*state.StateNode) error {
```

Comment:
1. Can we add a comment to the method that documents that the `addTaint` bool parameter is overloaded to "add if true" or "remove if false"? Maybe `requireTaint` is a better name?
2. Do we want to make this a general func that takes a taint as input, or do we want this to be *just* for enforcing the `v1beta1.DisruptingNoScheduleTaint` taint config specifically? If the latter, should we maybe rename the method to something like `disruptingNoScheduleTaintCandidates` (I know that's a horribly long name).